dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/dataitem/DataItemQueryController.java

```diff
@@ -0,0 +1,200 @@
+     */
+    @GetMapping( value = API_RESOURCE_PATH + ".xml", produces = APPLICATION_XML_VALUE )
+    public ResponseEntity<RootNode> getXml( @RequestParam
+    final Map<String, String> urlParameters, final OrderParams orderParams, final User currentUser )
+    {
+        log.debug( "Looking for data items (XML response)" );
+
+        return getDimensionalItems( currentUser, urlParameters, orderParams );
+    }
+
+    /**
+     * Based on the informed arguments, this method will read the URL and based on
+     * the give params will retrieve the respective data items.
+     * 
+     * @param currentUser the logged user
+     * @param urlParameters the request url params
+     * @param orderParams the request order params
+     * @return the complete root node
+     */
+    private ResponseEntity<RootNode> getDimensionalItems( final User currentUser,
+        final Map<String, String> urlParameters,
+        final OrderParams orderParams )
+    {
+        // Defining the input params.
+        final List<String> fields = newArrayList( contextService.getParameterValues( FIELDS ) );
+        final List<String> filters = newArrayList( contextService.getParameterValues( FILTER ) );
+        final WebOptions options = new WebOptions( urlParameters );
+
+        // checkPaginationSupport( filters );
+
+        if ( fields.isEmpty() )
+        {
+            fields.addAll( Preset.ALL.getFields() );
+        }
+
+        // Extracting the target entities to be queried.
+        final Set<Class<? extends BaseDimensionalItemObject>> targetEntities = dataItemServiceFacade
+            .extractTargetEntities( filters );
+
+        // Checking if the user can read all the target entities.
+        checkAuthorization( currentUser, targetEntities );
+
+        // Retrieving the data items based on the input params.
+        final List<BaseDimensionalItemObject> dimensionalItems = dataItemServiceFacade.retrieveDataItemEntities(
+            targetEntities, filters, options, orderParams );
+
+        // Creating the response node.
+        final RootNode rootNode = createMetadata();
+        responseHandler.addResultsToNode( rootNode, dimensionalItems, fields );
+        responseHandler.addPaginationToNode( rootNode, new ArrayList<>( targetEntities ), currentUser, options,
+            filters );
+
+        if ( isNotEmpty( dimensionalItems ) )
+        {
+            return new ResponseEntity<>( rootNode, FOUND );
+        }
+
+        return new ResponseEntity<>( rootNode, NOT_FOUND );
+    }
+
+    private void checkAuthorization( final User currentUser,
+        final Set<Class<? extends BaseDimensionalItemObject>> entities )
+    {
+        if ( isNotEmpty( entities ) )
+        {
+            for ( final Class<? extends BaseDimensionalItemObject> entity : entities )
+            {
+                if ( !aclService.canRead( currentUser, entity ) )
+                {
+                    throw new IllegalQueryException(
```

Comment:
Just a consideration: do we need to fail the entire request if one of the entity is not readable?

Reply:
I'm not sure. In the other hand, if we don't fail the user might have the impression that results are missing as we will skip the non-readable ones behind the scenes. This seems to be a behaviour that should come from the requirements.

@larshelge, what's your opinion on this situation? Luciano raised a good point.

Thanks

Reply:
I think we should simply filter out the items which the user does not have access to. That is how it works for other APIs.